tests/chunk.cpp

```diff
@@ -572,6 +587,21 @@ TEST(Chunk2DTest, SliceDomainY)
     }
 }
 
+TEST(Chunk2DTest, SliceDomainXOutOfBounds)
+{
+#if !defined(NDEBUG) // The assertion is only checked if NDEBUG isn't defined
+    ChunkXY<double> chunk(dom_x_y.remove(DVectXY(1, 0), DVectXY(1, 0)));
+
+    char const* const death_msg
+            = R"rgx([Aa]ssert.*DiscreteDomain<QueryDDims...>\(this->m_domain\).contains\(odomain.front\(\)\) && DiscreteDomain<QueryDDims...>\(this->m_domain\).contains\(odomain.back\(\)\))rgx";
+
+    EXPECT_DEATH(chunk[dom_x_y.remove_last(DVectXY(1, 0))], death_msg);
+    EXPECT_DEATH(chunk[dom_x_y.remove_first(DVectXY(1, 0))], death_msg);
```

Comment:
Could you please illustrate or comment this example in detail?
For me, it is not clear what has been actually removed from each domain.
It would be useful if you describe

1. what is the original scope of `dom_x_y`
2. what is the scope of chunk after remove (l. 593)
3. what is the scope of last-removed `dom_x_y` (l. 598)
4. what is the scope of first-removed dom_x_y (l. 599)

Reply:
thanks!